src/new/wallet.js

```diff
@@ -1201,6 +1204,8 @@ class HathorWallet extends EventEmitter {
   async onTxArrived(tx, isNew) {
     const tokensHistory = await this.getPreProcessedData('historyByToken');
     const tokensBalance = await this.getPreProcessedData('balanceByToken');
+    const updatedTokensBalance = this._getBalanceRaw();
```

Comment:
Why do we have to recalculate all balances every new transaction? Can't we just update the values?